query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -144,9 +145,15 @@ private String convertFilterToString(
     } else {
       switch (filter.getOperator()) {
         case LIKE:
-          // The like operation in PQL looks like `regexp_like(lhs, rhs)`
+          /**
+           * If the text index is not enabled on lhs expression, - the pql looks like
+           * `regexp_like(lhs, rhs)` else - the pql looks like `text_match(lhs, rhs)`
+           */
+          operator = handleLikeOperatorConversion(filter.getLhs());
           Expression rhs =
               handleValueConversionForLiteralExpression(filter.getLhs(), filter.getRhs());
+          rhs = postProcessValueConversionForLikeOperator(operator, rhs);
```

Comment:
nit: rather than taking multiple steps to calculate the RHS in the top level, could you do something like
```java
          operator = handleLikeOperatorConversion(filter.getLhs());
          Expression rhs =
              handleValueConversionForLikeArgument(filter.getLhs(), filter.getRhs());
```

and then have `handleValueConversionForLikeArgument` do the delegation.

Reply:
Done.